app/assets/javascripts/financial_transactions/new.js

```diff
@@ -1,17 +1,17 @@
 $(document).ready(function() {
   $('#contact_email').change(function (){
-    var email_regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/i;
+    var email_regex = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i;
     if(!email_regex.test($('#contact_email').val()))
       $('#contact_email').parent().parent().addClass('has-error');
-    else 
+    else
       $('#contact_email').parent().parent().removeClass('has-error');
   });
 
   $('#contact_phone').change(function (){
     var email_regex = /^[0-9]{10}$/i;
```

Comment:
These vars should be 'phone_regex' right? Also I wonder if it would be better to look for formatted phone numbers here and not just a string of 10 digits. Perhaps a regex like the following:
`/\(?[0-9]{3}\)?-[0-9]{3}-[0-9]{4}/` which looks for xxx-xxx-xxxx format but also allows optional parenthesis (if we allow optional parenthesis we should add a note to the field or to the placeholders alerting the user they can user parenthesis if they wish). Jquery validator may also be able to handle this as @Anbranin mentioned. And as I said in another comment, perhaps allow more than 10 digits to account for non-US numbers also

Reply:
I'm on board with all of this.
I'm just going to scrap this function.  Currently, all it does is give the input box a red outline if the input doesn't match this regex. It doesn't actually stop the user from submitting even if the format doesn't match.  Since I'm going to make a change to allow for various format inputs, the regex would be unnecessarily complicated.